cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -269,6 +269,10 @@ std::unique_ptr<cufile_input_impl> make_cufile_input(std::string const& filepath
       if (cufile_integration::is_always_enabled()) throw;
     }
   }
+#else
```

Comment:
Is the try-catch still required? 
Perhaps whatever is thrown in `std::make_unique<cufile_input_impl>(filepath)` should just pass through to the caller.